client/wfnews-war/src/main/angular/src/app/services/report-of-fire-service.ts

```
@@ -1,169 +1,235 @@
-        }
+  constructor(private appConfigService: AppConfigService,
+    private commonUtilityService: CommonUtilityService,
+    private storage: Storage) { }
 
-      }
+  submittedOffline: boolean
+  longitude: number
+  latitude: number
+
+  async saveReportOfFire(reportOfFire: ReportOfFireType): Promise<any> {
+
+    let rofUrl = this.appConfigService.getConfig().rest['fire-report-api']
+    let resource = JSON.stringify(reportOfFire)
+
+    this.latitude = reportOfFire.fireLocation[0];
+    this.longitude = reportOfFire.fireLocation[1];
 
-    async blobToBase64 (url): Promise<string> {
-      return new Promise(async (resolve, _) => {
-        // do a request to the blob uri
-        const response = await fetch(url);
-      
-        // response has a method called .blob() to get the blob file
-        const blob = await response.blob();
-      
-        // instantiate a file reader
-        const fileReader = new FileReader();
-      
-        // read the file
-        fileReader.readAsDataURL(blob);
-      
-        fileReader.onloadend = function(){
-          resolve(fileReader.result as string); // Here is the base64 string
+    try {
+      const formData = new FormData()
+      formData.append('resource', resource)
+
+      if (reportOfFire.image1 !== null && reportOfFire.image1 !== undefined && reportOfFire.image1) formData.append('image1', await this.convertToBase64(reportOfFire.image1))
+      if (reportOfFire.image2 !== null && reportOfFire.image2 !== undefined && reportOfFire.image2) formData.append('image2', await this.convertToBase64(reportOfFire.image2))
+      if (reportOfFire.image3 !== null && reportOfFire.image3 !== undefined && reportOfFire.image3) formData.append('image3', await this.convertToBase64(reportOfFire.image3))
+
+      // if the device is offline save RoF in storage
+      await (this.commonUtilityService.checkOnlineStatus().then(result => {
```

Add a catch for the promise so we don't get a silent fail. You can re-throw to the surrounding catch, but it will likely be de-scoped by the time the promise is done

---

Try-catch added